### PR TITLE
Compiler - Set AMD clang++ as default compiler for HIP

### PR DESCRIPTION
**Pull request by kiritigowda**
* Set AMD Clang++ as default compiler for the project
* If No ROCm found use system defaults
* Doc fixes
* Minor CMakeList fixes

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -18,6 +18,21 @@
...
+# ROCM Path
+if(DEFINED ENV{ROCM_PATH})
+  set(ROCM_PATH $ENV{ROCM_PATH} CACHE PATH "Default ROCm installation path")
+elseif(ROCM_PATH)
+  message("-- INFO:ROCM_PATH Set -- ${ROCM_PATH}")
+else()
+  set(ROCM_PATH /opt/rocm CACHE PATH "Default ROCm installation path")
+endif()
+# Set AMD Clang as default compiler
+if (NOT DEFINED CMAKE_CXX_COMPILER AND EXISTS "${ROCM_PATH}//bin/amdclang++")
```
Comment (LakshmiKumar23) [inline CMakeLists.txt:31]: remove additional `/`

File: model_compiler/python/nnir_to_zendnn.py [hunk 2]
```diff
@@ -93,6 +93,19 @@ def generateCMakeFiles(graph,outputFolder):
...
+# ROCM Path
+if(DEFINED ENV{ROCM_PATH})
+    set(ROCM_PATH $ENV{ROCM_PATH} CACHE PATH "Default ROCm installation path")
+elseif(ROCM_PATH)
+    message("-- INFO:ROCM_PATH Set -- ${ROCM_PATH}")
+else()
+    set(ROCM_PATH /opt/rocm CACHE PATH "Default ROCm installation path")
+endif()
+# Set AMD Clang as default compiler
+if (NOT DEFINED CMAKE_CXX_COMPILER AND EXISTS "${ROCM_PATH}//bin/amdclang++")
```
Comment (LakshmiKumar23) [inline model_compiler/python/nnir_to_zendnn.py:105]: remove additional `\`

File: tests/zen_dnn_tests/conv/CMakeLists.txt [hunk 3]
```diff
@@ -24,6 +24,19 @@
...
+# ROCM Path
+if(DEFINED ENV{ROCM_PATH})
+    set(ROCM_PATH $ENV{ROCM_PATH} CACHE PATH "Default ROCm installation path")
+elseif(ROCM_PATH)
+    message("-- INFO:ROCM_PATH Set -- ${ROCM_PATH}")
+else()
+    set(ROCM_PATH /opt/rocm CACHE PATH "Default ROCm installation path")
+endif()
+# Set AMD Clang as default compiler
+if (NOT DEFINED CMAKE_CXX_COMPILER AND EXISTS "${ROCM_PATH}//bin/amdclang++")
```
Comment (LakshmiKumar23) [inline tests/zen_dnn_tests/conv/CMakeLists.txt:36]: remove additional `\`

File: tests/zen_dnn_tests/mnist/CMakeLists.txt [hunk 4]
```diff
@@ -25,6 +25,19 @@
...
+# ROCM Path
+if(DEFINED ENV{ROCM_PATH})
+    set(ROCM_PATH $ENV{ROCM_PATH} CACHE PATH "Default ROCm installation path")
+elseif(ROCM_PATH)
+    message("-- INFO:ROCM_PATH Set -- ${ROCM_PATH}")
+else()
+    set(ROCM_PATH /opt/rocm CACHE PATH "Default ROCm installation path")
+endif()
+# Set AMD Clang as default compiler
+if (NOT DEFINED CMAKE_CXX_COMPILER AND EXISTS "${ROCM_PATH}//bin/amdclang++")
```
Comment (LakshmiKumar23) [inline tests/zen_dnn_tests/mnist/CMakeLists.txt:37]: remove additional `\`

File: tests/zen_dnn_tests/mnist_app/CMakeLists.txt [hunk 5]
```diff
@@ -25,6 +25,19 @@
...
+# ROCM Path
+if(DEFINED ENV{ROCM_PATH})
+    set(ROCM_PATH $ENV{ROCM_PATH} CACHE PATH "Default ROCm installation path")
+elseif(ROCM_PATH)
+    message("-- INFO:ROCM_PATH Set -- ${ROCM_PATH}")
+else()
+    set(ROCM_PATH /opt/rocm CACHE PATH "Default ROCm installation path")
+endif()
+# Set AMD Clang as default compiler
+if (NOT DEFINED CMAKE_CXX_COMPILER AND EXISTS "${ROCM_PATH}//bin/amdclang++")
```
Comment (LakshmiKumar23) [inline tests/zen_dnn_tests/mnist_app/CMakeLists.txt:37]: same as above

**Review by LakshmiKumar23 — APPROVED** [review]

**Comment by LakshmiKumar23** [comment]
@kiritigowda Can be merged after all tests pass

**Comment by kiritigowda** [comment]
Passing all tests
```
Start  1: vx_core_test
 1/26 Test  #1: vx_core_test .......................   Passed    2.38 sec
      Start  2: vx_amd_media_test
 2/26 Test  #2: vx_amd_media_test ..................   Passed    0.50 sec
      Start  3: vx_nn_test
 3/26 Test  #3: vx_nn_test .........................   Passed    4.43 sec
      Start  4: vx_opencv_test
 4/26 Test  #4: vx_opencv_test .....................   Passed    0.33 sec
      Start  5: vx_rpp_test
 5/26 Test  #5: vx_rpp_test ........................   Passed    0.37 sec
      Start  6: runvx_test
 6/26 Test  #6: runvx_test .........................   Passed    0.74 sec
      Start  7: openvx_canny
 7/26 Test  #7: openvx_canny .......................   Passed    1.87 sec
      Start  8: openvx_channel_extract
 8/26 Test  #8: openvx_channel_extract .............   Passed    1.33 sec
      Start  9: openvx_color_convert
 9/26 Test  #9: openvx_color_convert ...............   Passed    1.32 sec
      Start 10: openvx_canny_CPU
10/26 Test #10: openvx_canny_CPU ...................   Passed    0.15 sec
      Start 11: openvx_channel_extract_CPU
11/26 Test #11: openvx_channel_extract_CPU .........   Passed    0.54 sec
      Start 12: openvx_color_convert_CPU
12/26 Test #12: openvx_color_convert_CPU ...........   Passed    0.14 sec
      Start 13: openvx_tests_runVisionPython_CPU
13/26 Test #13: openvx_tests_runVisionPython_CPU ...   Passed  109.02 sec
      Start 14: openvx_canny_GPU
14/26 Test #14: openvx_canny_GPU ...................   Passed    0.54 sec
      Start 15: openvx_channel_extract_GPU
15/26 Test #15: openvx_channel_extract_GPU .........   Passed    0.53 sec
      Start 16: openvx_color_convert_GPU
16/26 Test #16: openvx_color_convert_GPU ...........   Passed    0.55 sec
      Start 17: openvx_tests_runVisionPython_GPU
17/26 Test #17: openvx_tests_runVisionPython_GPU ...   Passed  158.77 sec
      Start 18: caffe2nnir2openvx_no_fuse
18/26 Test #18: caffe2nnir2openvx_no_fuse ..........   Passed  185.38 sec
      Start 19: caffe2nnir2openvx_fuse
19/26 Test #19: caffe2nnir2openvx_fuse .............   Passed  105.19 sec
      Start 20: caffe2nnir2openvx_fp16
20/26 Test #20: caffe2nnir2openvx_fp16 .............   Passed  104.76 sec
      Start 21: onnx2nnir2openvxx_no_fuse
21/26 Test #21: onnx2nnir2openvxx_no_fuse ..........   Passed  110.76 sec
      Start 22: onnx2nnir2openvxx_fuse
22/26 Test #22: onnx2nnir2openvxx_fuse .............   Passed  110.65 sec
      Start 23: onnx2nnir2openvxx_fp16
23/26 Test #23: onnx2nnir2openvxx_fp16 .............   Passed  109.43 sec
      Start 24: nnef2nnir2openvxx_no_fuse
24/26 Test #24: nnef2nnir2openvxx_no_fuse ..........   Passed  107.45 sec
      Start 25: nnef2nnir2openvxx_fuse
25/26 Test #25: nnef2nnir2openvxx_fuse .............   Passed  265.08 sec
      Start 26: nnef2nnir2openvxx_fp16
26/26 Test #26: nnef2nnir2openvxx_fp16 .............   Passed  122.33 sec

100% tests passed, 0 tests failed out of 26

Total Test time (real) = 1504.81 sec
```

**Pull request merged**